README.md

```diff
@@ -1,89 +1,162 @@
+implementing neural network operations.
+
+SYCL-BLAS has been developed in collaboration with the 
+*High Performance Computing & Architectures (HPCA) group*
+from the Universitat Jaume I [UJI](http://www.hpca.uji.es/).
+
+The project is maintained by [Codeplay Software](https://developer.codeplay.com).
+
+## Project Requirements
+
+SYCL-BLAS is designed to work with any SYCL 1.2.1 implementation.
+* We do not use any OpenCL interoperability so the code is pure C++.
+* The project is developed using [ComputeCpp](https://developer.codeplay.com)
+ using Ubuntu 16.04 on Intel CPU and Intel GPU.
+* A BLAS library, such as OpenBLAS, is also required by the project to verify the test results
+
+
+## Supported Platforms
+
+The master branch of SYCL-BLAS is regularly tested with the "Supported" hardware 
+listed on [the ComputeCpp Supported Platforms page](https://developer.codeplay.com/products/computecpp/ce/guides/platform-support).
+
+SYCL-BLAS may also work on other hardware and platforms assuming they implement 
+ SPIR or SPIR-V support.
+
+## Getting Started with SYCL-BLAS
+
+### Pre-requisites
+
+* CMake (version 3.4.2 and above)
+* OpenCL 1.2-capable hardware and drivers with SPIR 1.2 or SPIR-V support
+* OpenCL ICD Loader
+* OpenCL headers
+* gcc (version 5.4 and above)
+* [ComputeCpp](https://developer.codeplay.com)
+
+### Building SYCL-BLAS
+
+First get OpenBLAS. Instructions on building and installing this are on 
+the [project website](https://www.openblas.net/).
+
+```bash
+git clone https://github.com/xianyi/OpenBLAS.git
+make
+make install
+```
+
+SYCL-BLAS uses CMake as its build system. All the configuration options 
+can be found in the main CMakeLists.txt for the project and will show 
+up in CMake GUI if you use it.
+
+You will need to provide the location of the ComputeCpp install you are
+using with the variable `ComputeCpp_DIR`. It should point to the folder
+where `bin/`, `lib/` etc. are. This should be the only argument that is
+mandatory, everything else should be optional. The default build type is
+Release, though this can be overridden.
+
+The following command shows how to compile SYCL-BLAS. If you have installed OpenBLAS 
+in a different location adapt the path
+
+* set `ComputeCpp_DIR` to the ComputeCpp root path
+
+You may also need to tell CMake where to find the OpenBLAS binaries using the 
+```-DCMAKE_PREFIX_PATH=/opt/OpenBLAS``` argument
+
+```bash
+git clone --recursive  https://github.com/codeplaysoftware/sycl-blas.git
+mkdir build && cd build
+cmake ../ -DComputeCpp_DIR=/path/to/computecpp
+make
+```
+
+To install the SYCL-BLAS library
+
+```bash
+make install
+```
+
+Doxygen documentation can be generated by running:
+
+```bash
+doxygen doc/Doxyfile
+```
+
+### Sample Code
+
+The "samples" directory contains some sample code using SYCL-BLAS 
+ to implement GEMM and GEMV operations.
+
+#### How to compile and run the samples
+
+The "samples" folder contains a basic CMake configuration file and a module to find
```

Comment:
The way that the sample cmake has been designed at the moment is to separate the sample folder from the main project. If we follow this approach it means that the sycl blas should be compiled ad build first, then the sample can be built independently by having its own build folder. So it should have its own readme.md. If we are going to make it to make it compile as part of the library we need to and a flag like -DSYCL_BLAS_ENABLE_SAMPLE as an option to the main CMakeLists.txt.